Tests/Integration/ConcentratorDeduplicationDataMessagesIntegrationTests.cs

```diff
@@ -268,7 +269,35 @@ LoRaRequest CreateOTAARequest(LoRaPayloadData payload, string station)
             this.loraABPDevice.AppSKey = TestKeys.CreateAppSessionKey(2);
 
             // act/assert
-            await ActAndAssert(request1, request2, this.loraABPDevice, null, expectedNumberOfBundlerCalls, null, expectedMessagesUp, expectedMessagesDown, null);
+            await ActAndAssert(requests, this.loraABPDevice, null, expectedNumberOfBundlerCalls, null, expectedMessagesUp, expectedMessagesDown, null);
+        }
+
+        [Theory]
+        [InlineData("11-11-11-11-11-11-11-11", "22-22-22-22-22-22-22-22", "33-33-33-33-33-33-33-33", "44-44-44-44-44-44-44-44", "55-55-55-55-55-55-55-55", DeduplicationMode.Mark, 5, 1)]
+        [InlineData("11-11-11-11-11-11-11-11", "22-22-22-22-22-22-22-22", "33-33-33-33-33-33-33-33", "44-44-44-44-44-44-44-44", "55-55-55-55-55-55-55-55", DeduplicationMode.None, 5, 1)]
+        public async Task When_More_Than_4_Concentrators_Messages_Should_Go_Upstream(
+            string station1,
```

Comment:
if you change this to string[] stations, you can validate the count against the MaxConfirmationResubmitCount without generating the warning.

Reply:
Makes sense, done in 9b9f7c7b62e30fe04a54451477f5078f3f46bf7e